src/lib/arch/IArchMultithread.h

```diff
@@ -69,16 +69,16 @@ class IArchMultithread : public IInterface
   Not all platforms support all signals.  Unsupported signals are
   ignored.
   */
-  enum ESignal
+  enum class ThreadSignal : uint8_t
   {
-    kINTERRUPT, //!< Interrupt (e.g. Ctrl+C)
-    kTERMINATE, //!< Terminate (e.g. Ctrl+Break)
-    kHANGUP,    //!< Hangup (SIGHUP)
-    kUSER,      //!< User (SIGUSR2)
+    Interrupt, //!< Interrupt (e.g. Ctrl+C)
+    Terminate, //!< Terminate (e.g. Ctrl+Break)
+    Hangup,    //!< Hangup (SIGHUP)
+    User,      //!< User (SIGUSR2)
     kNUM_SIGNALS
```

Comment:
How about we change this while we're here?
```suggestion
    kMaxSignals
```

Reply:
using `NumThreadSignals`